Software/Arduino/cryologger_gvt/cryologger_gvt.ino

```diff
@@ -157,158 +258,177 @@ struct struct_timer
+  DEBUG_PRINTLN(HARDWARE_VERSION);
+  DEBUG_PRINT("Datetime:");
+  printTab(2);
+  printDateTime();
+  DEBUG_PRINT("Battery Voltage:");
+  printTab(1);
+  DEBUG_PRINTLN(readBattery());
+  DEBUG_PRINT("Storage:");
+  printTab(2);
+  DEBUG_PRINT_DEC(sdUsedMB, 1);
+  DEBUG_PRINT(" / ");
+  DEBUG_PRINT_DEC(sdTotalMB, 1);
+  DEBUG_PRINTLN(" MB");
+  DEBUG_PRINT("File Count:");
+  printTab(2);
+  DEBUG_PRINTLN(sdFileCount);
+
+  // Display welcome messages and logging configuration on OLED.
+  //printSystemSettings();
   displayWelcome();
-
-  // Display logging configuration
   printLoggingSettings();
   displayLoggingMode();
+  displaySeasonalMode();
+  printGnssSettings();
 
-  // Configure devices
-  configureWdt();     // Configure and start Watchdog Timer (WDT)
-  configureSd();      // Configure microSD
-  configureGnss();    // Configure u-blox GNSS receiver
-  syncRtc();          // Acquire GNSS fix and sync RTC with GNSS
-  checkDate();        // Get current day of month
-  createDebugFile();  // Create debug log file
-  setInitialAlarm();  // Configure RTC and set initial alarm
+  // Configure additional devices and logging parameters.
+  syncRtc();          // Synchronize RTC with GNSS.
+  checkDate();        // Update the current date.
+  createDebugFile();  // Create a debug log file on SD.
+  setSleepAlarm();    // Set the RTC alarm based on operation mode.
 
-  DEBUG_PRINT("Info - Datetime "); printDateTime();
-  DEBUG_PRINT("Info - Initial alarm "); printAlarm();
+  DEBUG_PRINT("[Setup] Info: Datetime ");
+  printDateTime();
 
-  // Indicate completion of setup
+  // Indicate setup is complete.
   displaySetupComplete();
 }
 
 // ----------------------------------------------------------------------------
-// Loop
+// Main Loop
 // ----------------------------------------------------------------------------
-void loop()
-{
-  // Check if alarm flag is set
-  if (alarmFlag)
-  {
-    DEBUG_PRINT("Info - Alarm trigger "); printDateTime();
-
-    // Configure logging
-    readRtc();        // Get the RTC's alarm date and time
-    setAwakeAlarm();  // Set alarm to stop logging and go to sleep
-    getLogFileName(); // Get timestamped log file name
-
-    // Check opeation mode
-    if (operationMode != 3) // Skip if in continuous mode
-    {
-      // Configure devices
-      qwiicPowerOn();       // Enable power to Qwiic connector
-      peripheralPowerOn();  // Enable power to peripherals
-      resetOled();          // Configure OLED display
-      configureSd();        // Configure microSD
-      configureGnss();      // Configure u-blox GNSS
+void loop() {
+  // Process RTC alarm events.
+  if (alarmFlag) {
+    DEBUG_PRINT("[Main] Info: Alarm trigger ");
+    printDateTime();
+
+    // Update RTC and logging configuration.
+    readRtc();          // Refresh current RTC time.
+    setLoggingAlarm();  // Schedule the wake-up alarm (end of logging period).
+    getLogFileName();   // Generate a new log file name with a timestamp.
```

Comment:
If I understand correctly, mode `CONTINUOUS` logs data to the same file, rather than e.g. starting a new file each day. This will generate large files if logging continuously for months at a time. Have deployments demonstrated that this approach is robust, both for logging and in subsequently extracting the data for post-processing?

Reply:
In continuous mode, a new file is created daily at 00:00 UTC. This functionality was implemented after previous versions, which logged to a single large file, experienced issues with data loss. I've found the daily files very manageable to work with, and I've written several Python scripts to help automate the conversion and post-processing tasks.